javaslang/src-gen/main/java/javaslang/Tuple3.java

```diff
@@ -124,6 +124,17 @@ public T1 _1() {
     }
 
     /**
+     * Wither of the 1st element of this tuple.
+     *
+     * Note: The wither is NOT named {@code _1} due to possible ambiguity when using method reference (Tuple3::_1).
+     *
+     * @return a copy of this tuple with a new value for the 1st element of this Tuple.
+     */
+    public Tuple3<T1, T2, T3> w1(T1 t1) {
```

Comment:
We use meaningful names, the _1, _2, ... are one of the few exceptions in Javaslang.

Naming updates on immutable collections are an interesting topic - and very controversial discussed. We will use the names `set1`, `set2`, ... here.

Additionally we should also add `set(index, value)` to `Seq`!

---

In general there are many naming candidates for setting an element at an index.

* `replaceAt(index, value)` (would go adhere with the existing `replace` methods)
* `set(index, value)` (known from mutable collections)
* `updated` (like in Scala)
* `with` (looks like a builder DSL to me)
* ...

We already have `insert(index, element)`, `put(key, value)`, `replace(oldValue, newValue)`, ... These also might be confused with mutable operation - because we are used to mutable collections. If we would only know immutable collections, it would be crystal clear that `seq.set(index, value)` would return a new instance. Therefore there is really no reason to not take the shortest name `set`.